ticTacToe.js

```diff
@@ -1,6 +1,663 @@
 // wait for the DOM to finish loading
-$(document).ready(function{
+$(document).ready(function(){
+
+var gameBoard = [[[null], [null], [null]],
+				[[null], [null], [null]],
+				[[null], [null], [null]]];
+var x = 1;
+var o = 2;
+var turn = x;
+var winner = "none";
+
+//a function that sets all the squares to null
+function clearBoard() {
+	//loop through the rows
+    for(i = 0; i < gameBoard.length; i++) {
+    	//loop through each spot in the row
+        for(a = 0; a < gameBoard[i].length; a++) {
+        	//set the square to null
+            gameBoard[i][a] = null;
+
+        }
+    }
+    //clear all the squares
+    $('#board .row').children().text(' ');
+}
+
+//a function to check if the game is over and to end it
+function isGameOver() {
+	checkWinner();
+	//what to do if a player has won to announce the win and get ready for a new game
+	if(winner === "player X") {
+		clearBoard();
+		winner = "none";
+		turn = x;
+		alert("Player X has won the game!");
+		
+	} else if(winner === "player O") {
+		clearBoard();
+		winner = "none";
+		turn = x;
+		alert("Player O has won the game!");
+		
+	}
+}
+
+//a function to determine if there is a winner
+function checkWinner() {
```

Comment:
Could you have simplified this logic a bit?  
For example:

```js
for (var i=0; i < 3; i+=1;) {
  if( gameboard[i][0] === gameBoard[i][1] && gameBoard[i][1] === gameBoard[i][2]) {
    if (gameBoard[i][0] === 1) {
      winner = 'playerX';
    } 
    else {
      winner= "playerY"; 
   }
}
```
That handles just one set of possible wins, not all; but maybe you can figure out the rest.